browser/brave_stats/brave_stats_updater.cc

```diff
@@ -120,7 +120,7 @@ BraveStatsUpdater::BraveStatsUpdater(PrefService* pref_service)
 BraveStatsUpdater::~BraveStatsUpdater() {}
 
 void BraveStatsUpdater::OnProfileAdded(Profile* profile) {
-  if (profile == ProfileManager::GetPrimaryUserProfile()) {
```

Comment:
Wow, there is a comment that this method should not be used in desktop.
```
  // WARNING: do not use this function on Desktop platforms (Windows, Mac,
  // Linux). See https://crbug.com/1264436 for more info.
  // TODO(https://crbug.com/1264436): restrict this function to Android and
  // ChromeOS.
```
Is there any possibility to introduce regressions from this change?

Reply:
Wow- nice find, @spylogsster @simonhong. So we were using a bad method this whole time?

I think the re-creation of the deleted folder was the regression 😄

Reply:
`GetLastUsedProfile` has the same comment, but I think the fix is going to be to remove this completely and use the p3a data. That has been the plan for a while now and we have enough data now to show that the results match up pretty well.

Reply:
```// WARNING: if the profile does not exist, this function creates it
  // synchronously, causing blocking file I/O. Use GetLastUsedProfileIfLoaded()
  // to avoid creating the profile synchronously.
  static Profile* GetLastUsedProfile();```

Reply:
false alarm, it does not recreate the default profile

Reply:
Changed to use `GetLastUsedProfileIfLoaded`

Reply:
it will recreate the default profile if all profiles have been deleted so probably `GetLastUsedProfileIfLoaded` is the way to go because it should not have any side effects